java/ql/src/semmle/code/java/frameworks/JavaxJson.qll

```diff
@@ -8,55 +8,103 @@ private import semmle.code.java.dataflow.ExternalFlow
-        "javax.json;JsonObject;false;getBoolean;;;Argument[1];ReturnValue;value",
-        "javax.json;JsonObject;false;getInt;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getInt;;;Argument[1];ReturnValue;value",
-        "javax.json;JsonObject;false;getJsonArray;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getJsonNumber;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getJsonObject;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getJsonString;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getString;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonObject;false;getString;;;Argument[1];ReturnValue;value",
-        "javax.json;JsonObjectBuilder;false;add;;;Argument[-1];ReturnValue;value",
-        "javax.json;JsonObjectBuilder;false;add;;;Argument[1];Argument[-1];taint",
-        "javax.json;JsonObjectBuilder;false;addNull;;;Argument[-1];ReturnValue;value",
-        "javax.json;JsonObjectBuilder;false;build;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonReader;false;read;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonReader;false;readArray;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonReader;false;readObject;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonReaderFactory;false;createReader;;;Argument[0];ReturnValue;taint",
-        "javax.json;JsonString;false;getChars;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonString;false;getString;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonValue;false;toString;;;Argument[-1];ReturnValue;taint",
-        "javax.json;JsonWriter;false;write;;;Argument[0];Argument[-1];taint",
-        "javax.json;JsonWriter;false;writeArray;;;Argument[0];Argument[-1];taint",
-        "javax.json;JsonWriter;false;writeObject;;;Argument[0];Argument[-1];taint",
-        "javax.json;JsonWriterFactory;false;createWriter;;;Argument[-1];Argument[0];taint"
-      ]
+      ["javax", "jakarta"] + ".json;" +
+        [
+          "Json;false;createArrayBuilder;(JsonArray);;Argument[0];ReturnValue;taint",
+          "Json;false;createArrayBuilder;(Collection);;Element of Argument[0];ReturnValue;taint",
+          "Json;false;createDiff;;;Argument[0..1];ReturnValue;taint",
+          "Json;false;createMergeDiff;;;Argument[0..1];ReturnValue;taint",
+          "Json;false;createMergePatch;;;Argument[0];ReturnValue;taint",
+          "Json;false;createObjectBuilder;(JsonObject);;Argument[0];ReturnValue;taint",
+          "Json;false;createObjectBuilder;(Map);;MapKey of Argument[0];ReturnValue;taint",
+          "Json;false;createObjectBuilder;(Map);;MapValue of Argument[0];ReturnValue;taint",
+          "Json;false;createPatch;;;Argument[0];ReturnValue;taint",
+          "Json;false;createReader;;;Argument[0];ReturnValue;taint",
+          "Json;false;createWriter;;;Argument[0];ReturnValue;taint",
```

Comment:
In case you decide to add `JsonPatchBuilder` it would be good to add [`Json.createPatchBuilder(JsonArray)`](https://jakarta.ee/specifications/platform/9.1/apidocs/jakarta/json/json#createPatchBuilder-jakarta.json.JsonArray-) as well.